lib/history_provider.dart

```diff
@@ -7,6 +7,10 @@ import 'package:uuid/uuid.dart';
 
 import 'link_matcher.dart';
 
+part 'providers/browser_history.dart';
+part 'providers/hash_history.dart';
+part 'providers/memory_history.dart';
```

Comment:
By including all three history implementations in the same library, it introduces a browser dependency even when using the in-memory history provider. I'm fine with that since most consumer apps will have a browser dependency anyway, but something to keep in mind that could potentially be improved.